<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>View</class>
 <widget class="QMainWindow" name="View">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1436</width>
    <height>920</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1346</width>
    <height>819</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>View</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="s21::GLWidget" name="OGLWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>1024</width>
      <height>768</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>1024</width>
      <height>768</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>1024</width>
      <height>768</height>
     </size>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>10</y>
      <width>300</width>
      <height>768</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>300</width>
      <height>768</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>300</width>
      <height>768</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <widget class="QPushButton" name="pushButton_OpenFile">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>27</y>
       <width>121</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Open File</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_LOAD">
     <property name="geometry">
      <rect>
       <x>158</x>
       <y>604</y>
       <width>132</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Load Settings</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_SAVE">
     <property name="geometry">
      <rect>
       <x>11</x>
       <y>604</y>
       <width>133</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Save Settings</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_resetTransform">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>27</y>
       <width>121</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_defaultSettings">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>634</y>
       <width>281</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset to default settings</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_main">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>5</y>
       <width>281</width>
       <height>20</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="text">
      <string>Part 1. 3DViewer v2.0</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>281</width>
       <height>243</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_affine">
      <item row="5" column="0">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_MoveZ">
        <property name="singleStep">
         <double>0.250000000000000</double>
        </property>
        <property name="value">
         <double>0.250000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QPushButton" name="pushButton_MoveMinusY">
        <property name="text">
         <string>Move Y -</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_setY">
        <property name="maximum">
         <double>360.000000000000000</double>
        </property>
        <property name="value">
         <double>7.200000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QPushButton" name="pushButton_MovePlusY">
        <property name="text">
         <string>Move Y +</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_MoveY">
        <property name="singleStep">
         <double>0.250000000000000</double>
        </property>
        <property name="value">
         <double>0.250000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="pushButton_rotateMinusZ">
        <property name="text">
         <string>Rotate Z -</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="pushButton_rotatePlusY">
        <property name="text">
         <string>Rotate Y +</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QPushButton" name="pushButton_MovePlusX">
        <property name="text">
         <string>Move X +</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_MoveX">
        <property name="singleStep">
         <double>0.250000000000000</double>
        </property>
        <property name="value">
         <double>0.250000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QPushButton" name="pushButton_MoveMinusZ">
        <property name="text">
         <string>Move Z -</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="pushButton_rotatePlusZ">
        <property name="text">
         <string>Rotate Z +</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="pushButton_rotateMinusX">
        <property name="text">
         <string>Rotate X -</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="pushButton_MoveMinusX">
        <property name="text">
         <string>Move X -</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_setX">
        <property name="maximum">
         <double>360.000000000000000</double>
        </property>
        <property name="value">
         <double>7.200000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QPushButton" name="pushButton_MovePlusZ">
        <property name="text">
         <string>Move Z +</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="pushButton_rotatePlusX">
        <property name="text">
         <string>Rotate X +</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="pushButton_rotateMinusY">
        <property name="text">
         <string>Rotate Y -</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_setZ">
        <property name="maximum">
         <double>360.000000000000000</double>
        </property>
        <property name="value">
         <double>7.200000000000000</double>
        </property>
       </widget>
      </item>
      <item row="6" column="2">
       <widget class="QPushButton" name="pushButton_scalePlus">
        <property name="text">
         <string>Scale +</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QPushButton" name="pushButton_scaleMinus">
        <property name="text">
         <string>Scale -</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_Scale">
        <property name="minimum">
         <double>1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>2.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.100000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_record">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>673</y>
       <width>281</width>
       <height>20</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="text">
      <string>Part 3. Bonus. Record</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_ScreenSave">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>698</y>
       <width>131</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Save Screenshot</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_FileType">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>698</y>
       <width>80</width>
       <height>32</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>bmp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jpeg</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_settings">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>306</y>
       <width>281</width>
       <height>20</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="text">
      <string>Part 2. Bonus. Settings</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>11</x>
       <y>329</y>
       <width>281</width>
       <height>272</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_settings" columnstretch="0,1,0">
      <item row="0" column="0">
       <widget class="QLabel" name="label_proection">
        <property name="text">
         <string>Проекция: </string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="QComboBox" name="comboBox_ProType">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <item>
         <property name="text">
          <string>Центральная</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Паралельная</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_lineType">
        <property name="text">
         <string>Тип линии:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="2">
       <widget class="QComboBox" name="comboBox_LineType">
        <item>
         <property name="text">
          <string>Сплошная</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Пунктирная</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_lineWidth">
        <property name="text">
         <string>Толщина линии:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1" colspan="2">
       <widget class="QSpinBox" name="spinBox_LineWidth">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>64</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_vertexType">
        <property name="text">
         <string>Тип Вершины</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1" colspan="2">
       <widget class="QComboBox" name="comboBox_VertexType">
        <item>
         <property name="text">
          <string>Нет</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Квадрат</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Круг</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_Width">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Толщина вершины:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1" colspan="2">
       <widget class="QSpinBox" name="spinBox_VertexWidth">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>64</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_colorVertex">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Цвет вершин</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QWidget" name="widget_colorVretex" native="true">
        <property name="enabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QPushButton" name="pushButton_setColorVertex">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_corolLine">
        <property name="text">
         <string>Цвет линий</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QWidget" name="widget_colorLine" native="true"/>
      </item>
      <item row="6" column="2">
       <widget class="QPushButton" name="pushButton_setColor">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_color_BG">
        <property name="text">
         <string>Цвет фона</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QWidget" name="widget_BG" native="true"/>
      </item>
      <item row="7" column="2">
       <widget class="QPushButton" name="pushButton_setColorBG">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="pushButton_record">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>731</y>
       <width>91</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Record</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBox_autoRecording">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>736</y>
       <width>53</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Auto</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_saveGIF">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>731</y>
       <width>101</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Save to GIF</string>
     </property>
    </widget>
   </widget>
   <widget class="QComboBox" name="comboBox_objectDisplayType">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>790</y>
      <width>251</width>
      <height>32</height>
     </rect>
    </property>
    <property name="toolTipDuration">
     <number>-1</number>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="currentText">
     <string>Каркасная модель</string>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="maxVisibleItems">
     <number>3</number>
    </property>
    <item>
     <property name="text">
      <string>Каркасная модель</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Плоское затенение</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Мягкое затенение</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="pushButton_applyTexture">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>790</y>
      <width>171</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Нанести текстуру</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_offTexture">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>790</y>
      <width>171</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Снять текстуру</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelLightPosition">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>792</y>
      <width>201</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Позиция ист. освещения (x, y, z)</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_LightPosX">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>790</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_LightPosY">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>790</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_LightPosZ">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>790</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.000000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_ApplyLightPos">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>786</y>
      <width>111</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Apply Light pos.</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelLightColor">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>820</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Цвет ист. освещения</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelLightPower">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>850</y>
      <width>231</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Интенсивность ист. освещения</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_LightPower">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>845</y>
      <width>111</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>off Ligtning</string>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider_LightPower">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>850</y>
      <width>160</width>
      <height>25</height>
     </rect>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="value">
     <number>90</number>
    </property>
    <property name="sliderPosition">
     <number>90</number>
    </property>
    <property name="tracking">
     <bool>true</bool>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="widget_LightColor" native="true">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>820</y>
      <width>93</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_LightColor">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>816</y>
      <width>42</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_UV_Map">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>820</y>
      <width>171</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>UV Map</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1436</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">View/glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
